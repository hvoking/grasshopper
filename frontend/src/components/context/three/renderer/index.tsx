// React imports
import { useContext, createContext } from 'react';

// Third-party imports
import * as THREE from 'three';

const RendererContext: React.Context<any> = createContext(null)

export const useRenderer = () => {
	return (
		useContext(RendererContext)
	)
}

export const RendererProvider = ({children}: any) => {
	const createNewRenderer = (): any => {
		const renderer = new THREE.WebGLRenderer();
		renderer.setClearColor(0xd4d0c8);
		// renderer.setPixelRatio(window.devicePixelRatio)
		renderer.setSize( window.innerWidth, window.innerHeight - 178 );
		return renderer
	}

	return (
		<RendererContext.Provider value={{ createNewRenderer }}>
			{children}
		</RendererContext.Provider>
	)
}

RendererContext.displayName = "RendererContext";