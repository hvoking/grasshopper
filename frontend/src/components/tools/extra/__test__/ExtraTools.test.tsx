// React imports
import * as React from 'react';
import * as ReactDOM from 'react-dom';

// App imports
import { ExtraTools } from '../';

// Third-party imports
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup)
it("renders without crashing", () => {
	const div = document.createElement('div');
	ReactDOM.render(<ExtraTools></ExtraTools>, div)
})

it("renders Extra Tools correctly", () => {
	const {getByTestId} = render(<ExtraTools></ExtraTools>)
	expect(getByTestId('extraTools')).toHaveTextContent("#")

})

it("matches snapshot", () => {
	const tree = renderer.create(<ExtraTools></ExtraTools>).toJSON()
	expect(tree).toMatchSnapshot();
})