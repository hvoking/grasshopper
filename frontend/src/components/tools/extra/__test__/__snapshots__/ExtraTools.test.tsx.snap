// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="extraTools"
  data-testid="extraTools"
>
  <div>
    <div>
      #
    </div>
  </div>
  <div>
    <div>
      #
    </div>
  </div>
</div>
`;
